### ISSUE TITLE
[Bug] /v1/score returns validation error for single-label cross-encoder rerankers (e.g. bge-reranker-v2-m3)

### ISSUE THREAD

**Issue by ChengjieLi28** [sgl-project/sglang#38360]
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [ ] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Describe the bug

`POST /v1/score` fails with a pydantic validation error when serving a single-label cross-encoder reranker such as `BAAI/bge-reranker-v2-m3`:

```
2 validation errors for ScoringResponse scores.0 Input should be a valid list [type=list_type, input_value=5.875, input_type=float] scores.1 Input should be a valid list [type=list_type, input_value=-10.109375, input_type=float]
```

### Reproduction

```bash
python -m sglang.launch_server --model-path BAAI/bge-reranker-v2-m3 \
  --is-embedding --trust-remote-code --attention-backend triton

curl -s http://127.0.0.1:30000/v1/score -H "Content-Type: application/json" \
  -d '{"model": "bge-reranker-v2-m3", "query": "What is deep learning?",
       "items": ["Deep learning is a branch of machine learning.", "Nice weather today."]}'
```

Expected: {"scores": [[5.875], [-10.109375]], ...} — one row per item, matching the declared ScoringResponse.scores: List[List[float]].

Actual: validation error inside serving_score, returned to the client as an error response.

## Root cause
CrossEncodingPooler applies squeeze(-1) to the [num_labels] classification logits (python/sglang/srt/layers/pooler.py), so single-label heads emit a scalar per item. _process_single_item_scoring_results (tokenizer_manager_score_mixin.py) appends the scalar directly, producing a flat scores list that fails ScoringResponse construction.

The rerank endpoint already coerces this shape (_build_rerank_response, #16403); the score path was missed.

Reproduced on current main (df623d3cbd). Happy to submit a fix PR.

### Environment

```
W0908 01:22:45.163000 261546 miniconda3/envs/tcray_sgl/lib/python3.11/site-packages/torch/utils/_pytree.py:630] <enum 'KernelPreference'> is an Enum subclass and is now natively supported by torch.compile as an opaque value type. Calling register_constant() on Enum subclasses is deprecated and will be an error in a future release.
W0908 01:22:45.183000 261546 miniconda3/envs/tcray_sgl/lib/python3.11/site-packages/torch/utils/_pytree.py:630] <enum 'ScaleCalculationMode'> is an Enum subclass and is now natively supported by torch.compile as an opaque value type. Calling register_constant() on Enum subclasses is deprecated and will be an error in a future release.
Python: 3.11.16 (main, Aug 27 2026, 14:44:21) [GCC 14.3.0]
CUDA available: True
GPU 0,1: NVIDIA A10
GPU 0,1 Compute Capability: 8.6
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 13.0, V13.0.88
CUDA Driver Version: 580.126.20
PyTorch: 2.13.0+cu130
sglang: 0.5.18
sglang-kernel: 0.4.6.post1
flashinfer_python: 0.6.17
flashinfer_cubin: Module Not Found
flashinfer_jit_cache: Module Not Found
triton: 3.7.1
transformers: 5.12.1
torchao: 0.17.0
numpy: 2.3.5
aiohttp: 3.14.3
fastapi: 0.133.1
huggingface_hub: 1.29.0
interegular: 0.3.3
modelscope: 1.39.1
orjson: 3.12.0
outlines: 0.1.11
packaging: 26.3
psutil: 7.2.2
pydantic: 2.13.0
python-multipart: 0.0.32
pyzmq: 27.2.0
uvicorn: 0.52.4
uvloop: 0.22.1
vllm: Module Not Found
xgrammar: 0.2.1
openai: 2.6.1
tiktoken: 0.14.0
anthropic: 1.2.0
litellm: Module Not Found
torchcodec: 0.15.0
NVIDIA Topology: 
        GPU0    GPU1    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      PHB     0-55    0               N/A
GPU1    PHB      X      0-55    0               N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

Hypervisor vendor:: KVM
ulimit soft: 1024
```
